### PR TITLE
fix: test failures for logdna-agent

### PR DESCRIPTION
**Pull request by imprateeksh**
### Description

Some of the pipelines have failed due to failure in logdna. Below changes are done:
 * More logs are added.
 * Retry attempts are added.

#### Changes that affect the core Terraform module or submodules
- [x] Bug fix
- [ ] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: module-assets/ci/run-tests.sh [hunk 1]
```diff
@@ -127,9 +127,33 @@ if [ ${IS_PR} == true ]; then
...
+          if [ $RESULT_LOGDNA_START -eq 0 ]; then
+              echo "Logdna-agent started successfully"
+              break
+          else
+              echo "Logdna-agent start command exit status: $RESULT_LOGDNA_START"
+              echo "Logdna-agent Tag added: $REPO_NAME-PR$PR_NUM"
+              echo "Logdna-agent Status: $(systemctl status logdna-agent)"
+              LOGDNA_RUN_ATTEMPT=$((LOGDNA_RUN_ATTEMPT+1))
+              echo "=================================================== Logdna-agent: Service Log ==================================================="
+              grep -i error < /var/log/journal/logdna-agent.service.log | tail -n 10
```
[the remark this answers is not in the document]
Comment (imprateeksh) [reply module-assets/ci/run-tests.sh:150]: Done. printing last 20 lines without grepping "error"

File: module-assets/ci/run-tests.sh [hunk 2]
```diff
@@ -127,9 +127,33 @@ if [ ${IS_PR} == true ]; then
...
+          echo "Starting logdna-agent: [$LOGDNA_RUN_ATTEMPT/$MAX_RETRY_LOGDNA]"
+          systemctl start logdna-agent
+          RESULT_LOGDNA_START=$?
+
+          if [ $RESULT_LOGDNA_START -eq 0 ]; then
+              echo "Logdna-agent started successfully"
+              break
+          else
+              echo "Logdna-agent start command exit status: $RESULT_LOGDNA_START"
+              echo "Logdna-agent Tag added: $REPO_NAME-PR$PR_NUM"
```
[the remark this answers is not in the document]
Comment (imprateeksh) [reply module-assets/ci/run-tests.sh:146]: Resolved.

File: module-assets/ci/run-tests.sh [hunk 3]
```diff
@@ -127,9 +127,33 @@ if [ ${IS_PR} == true ]; then
...
+              echo "Logdna-agent Status: $(systemctl status logdna-agent)"
```
[the remark this answers is not in the document]
Comment (imprateeksh) [reply module-assets/ci/run-tests.sh:147]: Added `set +e` and `set -e` in this else block too.

**Comment by imprateeksh** [comment]
@ocofaigh - I have made the changes, can you please check once?

File: module-assets/ci/run-tests.sh [hunk 4]
```diff
@@ -127,9 +127,38 @@ if [ ${IS_PR} == true ]; then
...
+      ## Retry running logdna if fails to run and adding more logs
+
+      MAX_RETRY_LOGDNA=5
+      LOGDNA_RUN_ATTEMPT=1
+
+      while [ "$LOGDNA_RUN_ATTEMPT" -le "$MAX_RETRY_LOGDNA" ]; do
+          echo "Starting logdna-agent: [$LOGDNA_RUN_ATTEMPT/$MAX_RETRY_LOGDNA]"
+          set +e
+            systemctl start logdna-agent
+            RESULT_LOGDNA_START=$?
```
Comment (ocofaigh) [inline module-assets/ci/run-tests.sh:140]: no indentation should be used after `set +e`

File: module-assets/ci/run-tests.sh [hunk 5]
```diff
@@ -127,9 +127,38 @@ if [ ${IS_PR} == true ]; then
...
+              set +e
+                echo "Logdna-agent Status: $(systemctl status logdna-agent)"
+                LOGDNA_RUN_ATTEMPT=$((LOGDNA_RUN_ATTEMPT+1))
+                echo "=================================================== Logdna-agent: Service Log ==================================================="
+                  tail -n 20 /var/log/journal/logdna-agent.service.log
+                echo "================================================================================================================================="
+                if [ $LOGDNA_RUN_ATTEMPT -le $MAX_RETRY_LOGDNA ]; then
+                  echo "Retrying..."
+                fi
+              set -e
```
Comment (ocofaigh) [inline module-assets/ci/run-tests.sh:158]: no indentation should be used after `set +e`

File: module-assets/ci/run-tests.sh [hunk 6]
```diff
@@ -127,9 +127,38 @@ if [ ${IS_PR} == true ]; then
       export MZ_TAGS="$REPO_NAME-PR$PR_NUM"
       # Exclude extra logs
       export MZ_EXCLUSION_REGEX_RULES="/var/log/*"
-      echo "Starting logdna-agent"
-      systemctl start logdna-agent
-      echo "Logdna-agent Status: $(systemctl status logdna-agent)"
+
+      ## Retry running logdna if fails to run and adding more logs
+
+      MAX_RETRY_LOGDNA=5
```
Comment (ocofaigh) [inline module-assets/ci/run-tests.sh:133]: I would go with 3 retries. If the agent doesn't start after 3 theres probably a serious issue and we should fail

**Review by ocofaigh — CHANGES_REQUESTED** [review]
final comments

File: module-assets/ci/run-tests.sh (hunk 6 above)
Comment (imprateeksh) [reply module-assets/ci/run-tests.sh:133]: Ok will change this count to 3

File: module-assets/ci/run-tests.sh (hunk 4 above)
Comment (imprateeksh) [reply module-assets/ci/run-tests.sh:140]: sure. removed the indentation.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**